backends/aoti/aoti_partitioner.py

```diff
@@ -0,0 +1,76 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
```

Comment:
I am wondering if we could split the partitioner (or, more generally, any other part of this PR) to help the review/land/iterate process go faster